### PR TITLE
Sage Intacct Project Sync

### PR DESCRIPTION
**Pull request by KirtiGautam**

### CODE REVIEW

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 1]
```diff
@@ -110,7 +111,8 @@ export class ConfigurationComponent implements OnInit {
       that.isLoading = false;
       that.generalSettingsForm = that.formBuilder.group({
         reimburExpense: ['', Validators.required],
-        cccExpense: [null]
+        cccExpense: [null],
+        importProjects: [false],
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:115]: ,

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 2]
```diff
@@ -168,14 +178,14 @@ export class ConfigurationComponent implements OnInit {
...
-          that.settingsService.postGeneralSettings(that.workspaceId, reimbursableExpensesObject, cccExpensesObject)
+          that.settingsService.postGeneralSettings(that.workspaceId, reimbursableExpensesObject, cccExpensesObject, importProjects)
         ]
       ).subscribe(responses => {
         that.isLoading = true;
         that.snackBar.open('Configuration saved successfully');
-        that.router.navigateByUrl(`workspaces/${that.workspaceId}/dashboard`).then(() => {
-          that.windowReference.location.reload();
-        });
+        // that.router.navigateByUrl(`workspaces/${that.workspaceId}/dashboard`).then(() => {
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:186]: why is this commented?

Comment (KirtiGautam) [reply src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:186]: Ahh, I commented it for testing. Forgot to uncomment. I will do it right away.

File: src/app/core/services/mappings.service.ts [hunk 3]
```diff
@@ -231,7 +245,7 @@ export class MappingsService {
   getSageIntacctChargeCard() {
     const workspaceId = this.workspaceService.getWorkspaceId();
 
-    return this.apiService.get(`/workspaces/${workspaceId}/sage_intacct/charge_card_accounts/`, {});
+    return this.apiService.get(`/workspaces/${workspaceId}/sage_intacct/charge_card_accounts/`, {}); //item-> items/
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts:248]: ```//item-> items/```

Comment (KirtiGautam) [reply src/app/core/services/mappings.service.ts:248]: Yea, got it

File: src/app/core/services/mappings.service.ts [hunk 4]
```diff
@@ -298,15 +318,43 @@ export class MappingsService {
...
+    }), concatMap((res: any) => res.results),
+      reduce((arr, val) => {
+        arr.push(val);
+        return arr;
+      }, []));
   }
 
+  // Following is traditional method of recursive calls emplying two methods for recursive calls to API
+
+  // getAllMappings(sourceType): Observable<MappingsResponse[]> {
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts:342]: TODO: Shwetabh to decide what to finalise

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts (hunk 1 above)
Comment (ashwin1111) [reply src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:115]: remove ,

**Review by ashwin1111 — APPROVED** [review]

File: src/app/core/services/mappings.service.ts [hunk 5]
```diff
@@ -234,6 +248,12 @@ export class MappingsService {
     return this.apiService.get(`/workspaces/${workspaceId}/sage_intacct/charge_card_accounts/`, {});
   }
 
+  getSageIntacctItem() {
```
Comment (Shwetabhk) [inline src/app/core/services/mappings.service.ts:251]: `getSageIntacctItems()`

File: src/app/core/services/mappings.service.ts [hunk 6]
```diff
@@ -298,15 +318,43 @@ export class MappingsService {
     });
   }
 
-  getMappings(sourceType): Observable<MappingsResponse> {
+  getMappings(sourceType, uri = null): Observable<MappingsResponse> {
```
Comment (Shwetabhk) [inline src/app/core/services/mappings.service.ts:321]: Should there be spaces? `uri = null` or it should be `uri=null`
Just check once?

File: src/app/core/services/mappings.service.ts (hunk 4 above)
Comment (Shwetabhk) [reply src/app/core/services/mappings.service.ts:342]: Lets go with Kirti's approach

File: src/app/core/services/mappings.service.ts [hunk 7]
```diff
@@ -298,15 +318,43 @@ export class MappingsService {
...
+      }, []));
   }
 
+  // Following is traditional method of recursive calls emplying two methods for recursive calls to API
+
+  // getAllMappings(sourceType): Observable<MappingsResponse[]> {
+  //   return from(this.getAllMappingsInternal(sourceType));
+  // }
+
+  // private getAllMappingsInternal(sourceType, tempMappings = null, uri = null): Promise<MappingsResponse[]> {
```
Comment (Shwetabhk) [inline src/app/core/services/mappings.service.ts:346]: Let's remove all this commented code

File: src/app/si/dashboard/dashboard.component.ts [hunk 8]
```diff
@@ -174,8 +174,9 @@ export class DashboardComponent implements OnInit {
       that.mappingsService.postSageIntacctVendors(),
       that.mappingsService.postSageIntacctEmployees(),
       that.mappingsService.postSageIntacctAccounts(),
-      that.mappingsService.postSageIntacctExpensetypes()
-    ).subscribe(() => {
+      that.mappingsService.postSageIntacctExpensetypes(),
+      that.mappingsService.postSageIntacctItems()
+      ).subscribe(() => {
       // that.snackBar.open('All employee data synced from your fyle account');
```
Comment (Shwetabhk) [inline src/app/si/dashboard/dashboard.component.ts:180]: Lets remove the commented code

**Review by Shwetabhk — APPROVED** [review]

File: src/app/si/dashboard/dashboard.component.ts (hunk 8 above)
Comment (KirtiGautam) [reply src/app/si/dashboard/dashboard.component.ts]: Done

File: src/app/core/services/mappings.service.ts (hunk 7 above)
Comment (KirtiGautam) [reply src/app/core/services/mappings.service.ts]: Done

File: src/app/core/services/mappings.service.ts (hunk 6 above)
Comment (KirtiGautam) [reply src/app/core/services/mappings.service.ts:321]: Actually, it is supposed to be uri = null. Checked from [LINK](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

File: src/app/core/services/mappings.service.ts (hunk 5 above)
Comment (KirtiGautam) [reply src/app/core/services/mappings.service.ts]: Done.

**Pull request merged**